src/FSharpLint.Core/Framework/HintParser.fs

```diff
@@ -622,8 +623,7 @@ module HintParser =
         let private pminus: Parser<char, unit> = pchar '-'
 
         let private minusString (minus:char option, charList) =
-            if minus.IsSome then '-' :: charList else charList
-            |> charListToString
+            charListToString (if minus.IsSome then '-' :: charList else charList)
```

Comment:
@Mersho this is a bug in AvoidSinglePipeOperator rule actually (false positive), you can work on it (create failing test, etc) in a new branch